pkg/apis/app/v1alpha1/application_types.go

```diff
@@ -58,6 +58,9 @@ type ApplicationSpec struct {
 
 	// Notes contain a human readable snippets intended as a quick start for the users of the Application.
 	Notes string `json:"notes,omitempty"`
+
+	// Phase represents the current phase of the application.
+	Phase ApplicationPhase `json:"phase,omitempty"`
```

Comment:
Not sure we should use Phase in Spec. I think we may want to reuse this in Status. Also, no other types afaict use a Phase in the Spec.

We could put Phase in the Status, but then the client would be updating the Status, and this breaks with k8s convention. Also, based on what you are doing I agree that Spec is the right place for it.

If we are only ever going to have two phases though should we consider 
```go
ComponentsDeployed bool `json,"componentsDeployed,omitempty"`
```
or 
```go
Assembled bool `json,"assembled,omitempty"`
```
If we think there are going to be additional phases we could also, to disambiguate from typical Kubernetes Phases in Status objects, do something like
```go
AssemblyPhase ApplicationAssemblyPhase `json,"assemblyPhase,omitempty"`
```